core/src/utils/directive.ts

```diff
@@ -223,3 +225,80 @@ export const mergeDirectives =
 			},
 		};
 	};
+
+/**
+ * Properties for configuring server-side rendering directives.
+ */
+export interface AttributesDirectiveProps {
+	/**
+	 * Events to be attached to an HTML element.
+	 * @remarks
+	 * Key-value pairs where keys are event types and values are event handlers.
+	 */
+	events?: Partial<Record<keyof HTMLElementEventMap, (event: any) => void>>;
+
+	/**
+	 * Attributes to be added to the provided node.
+	 * @remarks
+	 * The `style` attribute must be added separately.
+	 */
+	attributes?: Record<string, AttributeValue | ReadableSignal<AttributeValue>>;
+
+	/**
+	 * Styles to be added to an HTML element.
+	 * @remarks
+	 * Key-value pairs where keys are CSS style properties and values are style values.
+	 */
+	styles?: Partial<Record<keyof CSSStyleDeclaration, StyleValue | ReadableSignal<StyleValue>>>;
+
+	/**
+	 * Class names to be added to an HTML element.
+	 * @remarks
+	 * Key-value pairs where keys are class names and values indicate whether the class should be added (true) or removed (false).
+	 */
+	classNames?: Record<string, boolean | ReadableSignal<boolean>>;
+}
+
+const getStore = <T>(value: T | ReadableSignal<T>) => (isStore(value) ? value : writable(value));
```

Comment:
Instead of using this function, you should use `toReadableStore` from `stores.ts`.
I think you only need a readable, not a writable. If you need a writable, `toWritableStore` exists as well in `stores.ts`.